oat-sa/extension-tao-proctoring
# Feature/tca 399/adjusted remaining time

Timer adjustments are included when calculating remaining time for proctor delivery execution list
 
Related to : https://oat-sa.atlassian.net/browse/TCA-399
 
During timer adjustment remaining time cache is updated in monitoring cache. In case extra time would be set as well (f.e. current `extended_time` LTI parameter is using it), then extra time recalculation is also including timer adjustment.
 
#### How to test
 
Manual testing can be performed in playground environment created for this feature or local environment with appropriate branches of `taoTests`, `taoQtiTest`, `taoProctoring`. The configuration option `DeliveryExecutionStateService::OPTION_TIME_HANDLING` should be set to `DeliveryExecutionStateService::TIME_HANDLING_TIMER_ADJUSTMENT`.

- As a test taker launch a proctored test with time limits
- As a proctor authorize the test to start, when the test is in progress pause it
- As a proctor refresh the delivery execution list
- As a proctor click on Change time button, specify how many minutes should be added/subtracted and click OK. Remaining time in the list should correspond to the adjustment.
  
#### Dependencies
   
Companion PR :
 - [ ] https://github.com/oat-sa/extension-tao-testqti/pull/1750
 - [ ] https://github.com/oat-sa/extension-tao-act/pull/1352
